lib/utils/import/simplenote/index.js

```diff
@@ -0,0 +1,59 @@
+import { EventEmitter } from 'events';
+import CoreImporter from '../';
+import { endsWith, isEmpty } from 'lodash';
+
+class SimplenoteImporter extends EventEmitter {
+  constructor({ noteBucket, tagBucket, options }) {
+    super();
+    this.noteBucket = noteBucket;
+    this.tagBucket = tagBucket;
+    this.options = options;
+  }
+
+  importNotes = filesArray => {
+    const coreImporter = new CoreImporter({
+      noteBucket: this.noteBucket,
+      tagBucket: this.tagBucket,
+    });
+
+    if (isEmpty(filesArray)) {
+      this.emit('status', 'error', 'No file to import.');
+      return;
+    }
+
+    const file = filesArray[0];
+
+    if (!endsWith(file.name.toLowerCase(), '.json')) {
+      this.emit('status', 'error', 'File name does not end in ".json".');
+      return;
+    }
+
+    // Limit file size we will read to 2mb
+    if (file.size > 2000000) {
```

Comment:
Today I tested with export files of varying sizes (80KB – 4MB). While large files did not choke the app in terms of memory, it did degrade the experience in non-trivial ways:

1. There is considerable lag between the point when all notes have been added to the noteBucket, and the point when the notes appear (all at once) in the Note List.
1. Clicking on notes in the Note List is unresponsive for a while. The `onClick` is firing, but the redux dispatch is being deferred until a certain point... I'm not immediately sure what is causing this.
1. It takes quite a while for all the notes to be synced over to the server, and it is almost impossible to tell when it has finished. The only way to tell is to try clicking the Log Out button and see if it throws an unsynced notes warning.
1. If there are lots of notes AND lots of tags, the tags take a long time to show up in the tag drawer.

With these constraints, I felt like a 500-note (~400KB) export file was as far as I wanted to go... 😵 And since these issues are occurring _after_ the CoreImporter is done, I'm guessing the other importers will encounter them too.

I'll see if there are some easy things we can do to address at least 1 and 2.

Reply:
^ Addressed in #986 #987 #988